### PR TITLE
[v2.9] Migrating serviceaccounts test from python to go

### PR DESCRIPTION
**Pull request by joesims22**
## Issue: 
[rancher/qa-tasks#1343](https://github.com/rancher/qa-tasks/issues/1343)
Note: Backports will be created once this gets approved

### CODE REVIEW

**Review by brudnak — APPROVED** [review]
lgtm!

File: tests/v2/validation/serviceaccounts/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+For the sa_test.go run, an RKE1 cluster is required.
+
+In your config file, set the following:
+```yaml
+rancher:
+  host: "rancher_server_address"
+  adminToken: "rancher_admin_token"
+  clusterName: "cluster_to_run_tests_on"
+  insecure: true/optional
+  cleanup: false/optional
```
Comment (Priyashetty17) [inline tests/v2/validation/serviceaccounts/README.md:12]: nit, can you change this to : 
`cleanup: true/optional`

File: tests/v2/validation/serviceaccounts/sa_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import (
+	"testing"
+	"strings"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+
+	"github.com/rancher/shepherd/extensions/serviceaccounts"
```
Comment (Priyashetty17) [inline tests/v2/validation/serviceaccounts/sa_test.go:13]: Looks like this extension is no longer available and has been moved to actions here: https://github.com/rancher/rancher/blob/main/tests/v2/actions/serviceaccounts/serviceaccounts.go 

Can you change this to: 
"github.com/rancher/rancher/tests/v2/actions/serviceaccounts"

**Review by Priyashetty17 — CHANGES_REQUESTED** [review]
Have a few minor comments

**Review by Priyashetty17 — APPROVED** [review]
lgtm

**Pull request merged**